whatshap/variants.py

```diff
@@ -149,7 +149,9 @@ def __init__(
     def n_paths(self) -> int:
         return len(self._paths)
 
-    def read(self, chromosome, variants, sample, reference, regions=None) -> ReadSet:
+    def read(
+        self, chromosome, variants, sample, reference, regions=None, valid_alleles=None
```

Comment:
Calling this `valid_alleles` is better. Thanks. Reading the code, there is the implicit assumption that valid_alleles has the same length as `variants`, correct? I would make this explicit by adding an `assert` and by mentioning in the docstring. Also, I would already mention that this needs to be a list of `Genotype` instance, because otherwise it will fail further down the line.

Reply:
I added type-hint and changed name to allowed_genotypes. Also added assert.